src/spells.cpp

```diff
@@ -837,11 +837,11 @@ bool InstantSpell::playerCastInstant(Player* player, std::string& param)
 	if (selfTarget) {
 		var.setNumber(player->getID());
 	} else if (needTarget || casterTargetOrDirection) {
-		Creature* target = nullptr;
+		std::shared_ptr<Creature> target = nullptr;
 		bool useDirection = false;
 
 		if (hasParam) {
-			Player* playerTarget = nullptr;
+			std::shared_ptr<Player> playerTarget = nullptr;
```

Comment:
Could this be `const auto`
```suggestion
			auto playerTarget = std::shared_ptr<Player>(nullptr);
```